### DISCUSSION TITLE
Bosch Roomthermostat II (BTH-RM) - Firmware update

### DISCUSSION THREAD

**Discussion by hmaiss** [Koenkk/zigbee2mqtt#20291] [Devices]
I have sniffed the firmware update for the Bosch Roomthermostat II (0x1209_0x300e_0x02066a30.ota) using whsniff, Wireshark and https://github.com/compujuckel/ZigbeeOtaExtractor. Now I am not able to upload the ota-file according to the following links

https://www.zigbee2mqtt.io/guide/usage/ota_updates.html
https://www.zigbee2mqtt.io/guide/configuration/ota-device-updates.html

The firmware update does not show up in the Z2M web interface.

Wenn I explicitly check the corresponding Device by publishing {"id": "0x001e5e0902929351"} to the MQTT-Topic zigbee2mqtt/bridge/request/device/ota_update/check I get the following MQTT response

{
  "message": "Device 'Arbeitszimmer/RAUMTHERMOSTAT' does not support OTA updates",
  "meta": {
    "device": "Arbeitszimmer/RAUMTHERMOSTAT",
    "status": "not_supported"
  },
  "type": "ota_update"
}

Is the firmware update feature currently not implemente with the Bosch Roomthermostat II (https://www.zigbee2mqtt.io/devices/BTH-RM.html) or is the problem at my side?

**Comment by dierochade** [comment]
I run several devices of this type and they do update. 
There is no need to manually add the files, they ota out of the box for me.

This is my current status:

Firmware build date
20231122
Firmware version
3.05.09
Manufacturer
[Bosch](https://www.zigbee2mqtt.io/supported-devices/#v=Bosch)
Model
[BTH-RA](https://www.zigbee2mqtt.io/devices/BTH-RA.html#bosch-bth-ra)

**Comment by hmaiss** [comment]
@dierochade

My question is concerned with BTH-RM and not with the BTH-RA.

In the meantime I am quite shure that the firmware update with the BTH-RM is not possible with Z2M, because it is not activated in the corresponding source code:

https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/bosch.ts

I would have tried to activate this feature in my Z2M Docker image. But there I find a correspoinding JS file, which IMHO basically does the same thing. The JS file is much more complex and activating the update is not so easy compared to the TS file above.
